packages/context-editor/src/components/ImageUploader.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { usePluginViewContext } from "@prosemirror-adapter/react";
+
+import type { FileUploadProps, FormattedFile } from "ui/customRenderers/fileUpload/fileUpload";
+import { FileUpload } from "ui/customRenderers/fileUpload/fileUpload";
+import { Label } from "ui/label";
+
+import type { ImageAttrs } from "../schemas/image";
+import { insertNodeAfterSelection } from "../utils/nodes";
+
+export type Upload = FileUploadProps["upload"];
+
+export const ImageUploader = ({ upload, onInsert }: { upload: Upload; onInsert: () => void }) => {
+	const { view } = usePluginViewContext();
+
+	const onUpload = (files: FormattedFile[]) => {
+		// Reverse the files since files are inserted starting at the selection
+		for (const file of files.reverse()) {
+			const attrs: ImageAttrs = {
+				src: file.fileUploadUrl || "",
+				id: "",
+				class: null,
+				alt: file.fileName,
+				linkTo: "",
+				credit: null,
+				license: null,
+				width: 100,
+				align: "center",
+			};
+			insertNodeAfterSelection(view.state, view.dispatch, "image", attrs);
```

Comment:
weirdly this seems to have different behavior in storybook compared to in the app. in storybook, the files insert in the opposite order, and also insert a node between each. 

in platform:
![image](https://github.com/user-attachments/assets/0e336a16-5471-477a-9922-fdf3f98c5693)

storybook:
<img width="394" alt="image" src="https://github.com/user-attachments/assets/87d949b5-c65d-4b14-bf3c-3c83db24ef35" />

Reply:
🤷 I don't think this should block your PR, but we'll see if it the behavior in prod is undesireable or non-deterministic